### PR TITLE
Centre on SANSLoad

### LINKED ISSUE #23960: SANS ISIS Backend add centering to SANSLoad
Currently the SANSLoad algorithm loads SANS data and applies calibrations if necessary. The instrument scientists would like it to additionally centre the workspaces here so that loaded data is calibrated and centered. 

The information about the centre position is already in the state object passed to SANSLoad. The centering is currently done as part of the processing here

thttps://github.com/mantidproject/mantid/blob/master/scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py#L353

Which should act as a good example. With this change we will technically be centering twice in a typical reduction. Once during loading and again during the reduction itself. It is possible the centering algorithm is clever enough for the second time to not take long if nothing has changed but this should be checked.

### PR DESCRIPTION
**Pull request by TTitcombe**
**Description of work.**
In SANS v2 GUI, workspaces are now centred on SANSLoad, before being moved.

Note that in SANSLoad the component property is not used, and the component is hard coded as "LAB". As outlined in #24529, currently both components are moved regardless of component.

This extra issue will be tackled in a separate PR, which will change SANSLoad back to using the component as provided as input.

**Report to:** nobody

**To test:**
1. Interfaces -> SANS -> SANS v2 GUI
2. Load a matching user and batch file from the attached data
3. Click the load button
4. In the ADS, expand the group workspace, and on each workspace right click and show instrument. 
5. In SANS -> Settings -> Mask click DisplayMask. 
Compare all of these images. 
Some workspaces from the ADS will show grey instruments, but the coloured ones should roughly match-up with the Displayed Mask. The most important part is that the centre of Display Mask and the Instruments matches

[PRdata.zip](https://github.com/mantidproject/mantid/files/2756090/PRdata.zip)

Fixes #23960 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/SANSLoadTest.py [hunk 1]
```diff
@@ -218,6 +217,47 @@ def _run_load(state, publish_to_cache, use_cached, move_workspace=False, beam_co
         # self.assertTrue(load_alg.isExecuted())
         return load_alg
 
+    def _check_that_sets_to_zero(self, workspace, move_info, comp_name=None):
```
Comment (Mantid-Matthew) [inline Testing/SystemTests/tests/analysis/SANSLoadTest.py]: Is this function used anywhere

Comment (TTitcombe) [reply Testing/SystemTests/tests/analysis/SANSLoadTest.py]: Removed the test which uses it but forgot to remove that function

**Comment by Mantid-Matthew** [comment]
When I attempt to load the LOQDEMO data I get the following error do to with moveing
image

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSLoad.py [hunk 2]
```diff
@@ -336,25 +328,34 @@ def set_property_with_number_of_workspaces(self, name, workspace_collection):
...
+                zero_alg.execute()
+                zeroed_workspace = zero_alg.getProperty("Workspace").value
+
+                # If beam centre was specified then use it
+                beam_coordinates = self.getProperty("BeamCoordinates").value
+                if beam_coordinates:
+                    move_alg.setProperty("BeamCoordinates", beam_coordinates)
+
+                # If component was specified then use it
+                component = self.getProperty("Component").value
```
[the remark this answers is not in the document]
Comment (TTitcombe) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSLoad.py]: Now executing algorithm once with HAB and once with LAB. These are coded explicitly, which should stop the error you mentioned below, where components other than LAB and HAB were specified

File: docs/source/algorithms/SANSLoad-v1.rst [hunk 3]
```diff
@@ -83,8 +83,8 @@ The *UseCached* setting will look for appropriate workspaces on the *AnalysisDat
 Move a workspace
 ~~~~~~~~~~~~~~~~
 
-It is possible to perform an initial, instrument-specific move of the selected component. Currently this move mechanism is implemented for **SANS2D**, **LOQ** and **LARMOR**. Other instruments will not be moved.
-If *MoveWorkspace* is selected, then a component and a beam position needs to be selected.
+The algorithm perform an initial, instrument-specific move of the selected component. Currently this move mechanism is implemented for **SANS2D**, **LOQ** and **LARMOR**. Other instruments will not be moved.
```
Comment (Mantid-Matthew) [inline docs/source/algorithms/SANSLoad-v1.rst]: Do we not want to add ZOOM to this instrument list?

Comment (TTitcombe) [reply docs/source/algorithms/SANSLoad-v1.rst]: ZOOM added now

**Review by Matthew-Andrew — APPROVED** [review]
All looks good to me

**Pull request merged**